libs/wingc/src/type_check.rs

```diff
@@ -3128,6 +3128,19 @@ impl<'a> TypeChecker<'a> {
 		for (inner_scope, ctx) in inner_scopes {
 			let scope = unsafe { &*inner_scope };
 			self.ctx = ctx;
+			for statement in &scope.statements {
```

Comment:
This logic feels a little out of place here and adds mostly unnecessary iteration. It should be possible to do the necessary check inside `type_check_struct` by verifying `env.parent` is `None`

If that doesn't work, it could be valuable enough to introduce something like `is_top_level` to ctx instead.

Reply:
Moved the typechecking to be at the struct level and there weren't any issues